metrics/api/src/main/java/io/helidon/metrics/api/AbstractRegistry.java

```diff
@@ -0,0 +1,791 @@
+    }
+
+    @Override
+    public SortedMap<MetricID, Histogram> getHistograms() {
+        return getHistograms(MetricFilter.ALL);
+    }
+
+    @Override
+    public SortedMap<MetricID, Histogram> getHistograms(MetricFilter filter) {
+        return getSortedMetrics(filter, Histogram.class);
+    }
+
+    @Override
+    public SortedMap<MetricID, Meter> getMeters() {
+        return getMeters(MetricFilter.ALL);
+    }
+
+    @Override
+    public SortedMap<MetricID, Meter> getMeters(MetricFilter filter) {
+        return getSortedMetrics(filter, Meter.class);
+    }
+
+    @Override
+    public SortedMap<MetricID, Timer> getTimers() {
+        return getTimers(MetricFilter.ALL);
+    }
+
+    @Override
+    public SortedMap<MetricID, Timer> getTimers(MetricFilter filter) {
+        return getSortedMetrics(filter, Timer.class);
+    }
+
+    @Override
+    public SortedMap<MetricID, ConcurrentGauge> getConcurrentGauges() {
+        return getConcurrentGauges(MetricFilter.ALL);
+    }
+
+    @Override
+    public SortedMap<MetricID, ConcurrentGauge> getConcurrentGauges(MetricFilter filter) {
+        return getSortedMetrics(filter, ConcurrentGauge.class);
+    }
+
+    @Override
+    public SortedMap<MetricID, SimpleTimer> getSimpleTimers() {
+        return getSimpleTimers(MetricFilter.ALL);
+    }
+
+    @Override
+    public SortedMap<MetricID, SimpleTimer> getSimpleTimers(MetricFilter filter) {
+        return getSortedMetrics(filter, SimpleTimer.class);
+    }
+
+    @Override
+    public Map<String, Metadata> getMetadata() {
+        return Collections.unmodifiableMap(allMetadata);
+    }
+
+    @Override
+    public Map<MetricID, Metric> getMetrics() {
+        return Collections.unmodifiableMap(allMetrics);
+    }
+
+    /**
+     * Access a metric by name. Used by FT library.
+     *
+     * @param metricName Metric name.
+     * @return Optional metric.
+     */
+    public Optional<Metric> getMetric(String metricName) {
+        return getOptionalMetricEntry(metricName).map(Map.Entry::getValue);
+    }
+
+    /**
+     * Returns type of this registry.
+     *
+     * @return The type.
+     */
+    public String type() {
+        return type.getName();
+    }
+
+    /**
+     * Determines if registry is empty.
+     *
+     * @return Outcome of test.
+     */
+    public boolean empty() {
+        return allMetrics.isEmpty();
+    }
+
+    @Override
+    public String toString() {
+        return type() + ": " + allMetrics.size() + " metrics";
+    }
+
+    /**
+     * Get internal map entry given a metric name. Synchronized for atomic access of more than
+     * one internal map.
+     *
+     * @param metricName The metric name.
+     * @return Optional map entry..
+     */
+    public synchronized Optional<Map.Entry<? extends Metric, List<MetricID>>> getOptionalMetricWithIDsEntry(String metricName) {
+        final List<MetricID> metricIDs = allMetricIDsByName.get(metricName);
```

Comment:
`allMetricIdsByName` is a concurrent structure; maybe use `computeIfAbsent` and remove `synchronized` from the method body (with care obviously)

Reply:
(Oh, I see the multiple accesses; hmm)

Reply:
Yes, accesses to multiple data structures, unfortunately. Luckily, registrations of new metrics and removals of existing ones are relatively rare, compared to look-ups, so the odds for contention are low.